elaichi/lib/app/failure.dart

```
@@ -0,0 +1,14 @@
+/// Gets error messages to the UI.
```

Can't we use this class for internal errors also?

---

Can you give an example use case? We can have a look.

---

Like local database replying to ViewModel about failures.

---

Yes, you can use it to do that. Wherever you want to throw an error, on catching an exception or any other reason, you can use the Failure class to throw an error. The error will have the same flow as the data on success and reach the ViewModel, and the ViewModel can have the functionality to take care of the subsequent UI response.